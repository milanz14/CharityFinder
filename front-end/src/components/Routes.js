import { Route, Switch, Redirect } from "react-router-dom";
import Organizations from "./Organizations";
import Home from "./Home";
import CategorySearch from "./CategorySearch";
import LocationSearch from "./LocationSearch";

const Routes = () => {
    return (
        <Switch>
            <Route exact path="/">
                <Home />
            </Route>
            <Route exact path="/organizations">
                <Organizations />
            </Route>
            <Route exact path="/categories">
                <CategorySearch />
            </Route>
            <Route exact path="/locations">
                <LocationSearch />
            </Route>
        </Switch>
    );
};

export default Routes;
