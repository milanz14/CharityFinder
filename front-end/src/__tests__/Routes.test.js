import React from "react";
import { render } from "@testing-library/react";
import Routes from "../components/Routes";
import { BrowserRouter } from "react-router-dom";

// props: Home, Organizations, CategorySearch

test("It renders without crashing", () => {
    render(
        <BrowserRouter>
            <Routes />
        </BrowserRouter>
    );
});

test("it matches the snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <Routes />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
