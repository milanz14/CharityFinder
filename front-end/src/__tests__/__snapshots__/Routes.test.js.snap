// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h1
      class="home-title"
    >
      Feeling Charitable?
    </h1>
    <a
      class="home-btn"
      data-testid="button"
      href="/organizations"
    >
      Let's Get Started
    </a>
  </div>
</DocumentFragment>
`;
